src/localCrud.js

```diff
@@ -0,0 +1,57 @@
+import { contactKey } from "./common";
+import Contacts from "./Contacts";
+import React from "react";
+
+function setLocalData(country, { comment, url, contacts, index } = {}) {
+    if (!country) {
+        console.log("No country, cannot set local data");
+        return null;
+    }
+
+    let countryData = {
+        comment: comment ?? "",
+        url: url ?? "",
+        contacts: contacts ?? {},
+        index: index ?? 0,
+    }
+    localStorage.setItem(country, JSON.stringify(countryData))
+}
+
+function addOrUpdateContact(countryKey, id, contact) {

```

Comment:
maybe this should be 2 functions, `createContact(countryKey, contact)` and `updateContact(countryKey, id, contact)`, that way, you can abstract away the index incrementation logic